unmarshal.go

```diff
@@ -82,7 +82,7 @@ func (d *decoder) populateStruct(structInstance interface{}, result map[string]i
 					field := refInfo.field(structValue)
 					assignValue(field, value)
 				}
-				if result != nil {
+				if result != nil && !d.options.excludeKnownFieldsFromMap {
```

Comment:
Changing the logic of the if statement will also affect the else logic. To prevent it I'd either move the new condition inside the if block or reverse the if else order

Reply:
fixed. moved the condition inside the if block